packages/shared/src/effect.ts

```diff
@@ -50,6 +50,45 @@ export const fetchEff = (
     }),
   );
 
+// TODO: rename the other one to fetchEffJsonSchema and this to fetchEffJson
+// though generally I would avoid XandY kind utils in favor of composition
```

Comment:
if you wanna do this change now feel free - else I can look into it later

Reply:
it's just a bit of shotgun surgery since it's used all over, I can two-stage it. Keep the original just as a composition of extracted 'primitives'